workflow-ui/compose/src/androidTest/java/com/squareup/workflow1/ui/compose/ComposeViewTreeIntegrationTest.kt

```diff
@@ -575,15 +577,15 @@ internal class ComposeViewTreeIntegrationTest {
     setRendering(BackStackScreen(EmptyRendering, backstack.asList()))
   }
 
-  data class TestModal(
+  data class TestOverlay(
     override val content: Screen
-  ) : ScreenOverlay<Screen>, AndroidOverlay<TestModal> {
-    override fun <ContentU : Screen> map(transform: (Screen) -> ContentU) = error("Not implemented")
+  ) : ScreenOverlay<Screen>, AndroidOverlay<TestOverlay> {
+    override fun <U : Screen> map(transform: (Screen) -> U) = error("Not implemented")
 
-    override val dialogFactory = object : ScreenOverlayDialogFactory<Screen, TestModal>(
-      TestModal::class
-    ) {
-    }
+    override val dialogFactory =
+      OverlayDialogFactory<TestOverlay> { initialRendering, initialEnvironment, context: Context ->
+        ComponentDialog(context).setContent(initialRendering, initialEnvironment)
+      }
```

Comment:
I see.

Reply:
I see now why you're removing ScreenOverlayDialogFactory (I think). Not needed.

Reply:
Somehow I convinced myself that the view should be created before the dialog, and that caused so many problems.